pkg/module/sync/runner.go

```diff
@@ -350,6 +330,41 @@ func generateModuleStatus(module *common.Module, existStatus *v1beta2.ModuleStat
 	return moduleStatus
 }
 
+func generateModuleStatusFromError(module *common.Module, existStatus *v1beta2.ModuleStatus) v1beta2.ModuleStatus {
+	switch {
+	case errors.Is(module.Template.Err, templatelookup.ErrTemplateUpdateNotAllowed):
+		newModuleStatus := existStatus.DeepCopy()
+		newModuleStatus.State = shared.StateWarning
+		newModuleStatus.Message = module.Template.Err.Error()
+		return *newModuleStatus
+	case errors.Is(module.Template.Err, moduletemplateinfolookup.ErrNoTemplatesInListResult):
+		return v1beta2.ModuleStatus{
+			Name:    module.ModuleName,
+			Channel: module.Template.DesiredChannel,
+			FQDN:    module.FQDN,
+			State:   shared.StateWarning,
+			Message: module.Template.Err.Error(),
+		}
+	case errors.Is(module.Template.Err, moduletemplateinfolookup.ErrWaitingForNextMaintenanceWindow):
+		newModuleStatus := existStatus.DeepCopy()
+		newModuleStatus.Message = module.Template.Err.Error()
+		return *newModuleStatus
+	case errors.Is(module.Template.Err, moduletemplateinfolookup.ErrFailedToDetermineIfMaintenanceWindowIsActive):
+		newModuleStatus := existStatus.DeepCopy()
+		newModuleStatus.Message = module.Template.Err.Error()
+		newModuleStatus.State = shared.StateError
+		return *newModuleStatus
```

Comment:
Note to reviewer: these are the two now conditions that have been added. The rest here is just moving the existing conditions to the new helper func. If not moving to helper func, linter complains that the func is too long.